src/schemas/generator/index.ts

```diff
@@ -13,12 +13,29 @@ export function migrate(generator: z.infer<typeof AnyGeneratorSchema>) {
     case '0':
       return migrate(v0.migrate(generator))
     case '1.0':
+      migrateEmptyExtractionModeInCorrelation(generator)
```

Comment:
after much thought I've opted for a migration to align the old `single` mode, migration system is really cool 🙌

Reply:
`migrate` is essentially a recursive function: once the file has been migrated to the latest version (currently 1.0`, it should just return the generator object.
If it requires extra migration, then it simply needs a new schema version. Otherwise once we add v2, it will look like this:
```ts
  switch (generator.version) {
    case '0':
      return v1.migrate(v0.migrate(generator))
    case '1.0':
      return migrate(migrateEmptyExtractionModeInCorrelation(generator))
    case '2':
      return generator
    default:
      return exhaustive(generator)
  }
```